wp-content/themes/pub/wporg-learn-2024/js/course-outline.js

```diff
@@ -0,0 +1,38 @@
+/* global wporgCourseOutlineData */
+
+document.addEventListener( 'DOMContentLoaded', function () {
+	const lessonData = wporgCourseOutlineData;
+
+	lessonData[ 'in-progress' ]?.forEach( function ( lesson ) {
+		const title = lesson.title;
+		const icon = lesson.icon;
+
+		const lessonLinks = document.querySelectorAll( '.wp-block-sensei-lms-course-outline-lesson' );
+
+		lessonLinks.forEach( function ( link ) {
+			const span = link.querySelector( 'span' );
+
+			if ( span && span.textContent.trim() === title ) {
```

Comment:
Currently using the lesson title to do the matching and replace the svg. The risk is that the Sensei class names might change.